src/ir_passes/mono/monosig.rs

```diff
@@ -222,25 +232,29 @@ impl MonoSig {
         if let Some(&idx) = self.param_map.find(p_idx) {
             idx
         } else {
+            let p_rep = ul.display(p_idx);
             // This param is a in a use site and should therefore have been found.
             let msg = match ul.get(p_idx).owner {
                 ir::ParamOwner::Loop => "let-bound parameter".to_string(),
                 ir::ParamOwner::Bundle(_) => {
                     "bundle-bound parameter".to_string()
                 }
                 ir::ParamOwner::Sig => "signature-bound parameter".to_string(),
-                ir::ParamOwner::Instance(inst) => format!(
+                ir::ParamOwner::Instance { inst, .. } => format!(
                     "parameter defined by instance `{}'",
                     ul.display(inst.ul())
                 ),
                 ir::ParamOwner::Exists => {
-                    "existentially quantified parameter".to_string()
+                    unreachable!(
+                        "existential parameter `{}' occurred in a use location",
+                        p_rep
+                    )
```

Comment:
Correct me if I'm wrong, but this implies that we can't use an existentially quantified parameters in any expressions?

Reply:
Should we not just use `get_exist_val` like we do elsewhere?

Reply:
We add the existential parameter to the binding when monomorphize the corresponding instance:
```
A := new Foo;  // A::Out binding added here
B := new Bar[A::Out];
```